File: verify_final.py
from playwright.sync_api import sync_playwright, expect
import time

def test_modern_features(page):
    page.on("console", lambda msg: print(f"CONSOLE: {msg.text}"))
    page.on("pageerror", lambda exc: print(f"PAGE ERROR: {exc}"))

    print("Navigating to Modern Dashboard...")
    page.goto("http://localhost:3000/modern")

    # Wait for the synchronizing overlay to disappear or timeout
    try:
        page.wait_for_selector("#loadingOverlay:not(.active)", timeout=15000)
    except:
        print("Timeout waiting for overlay to disappear")

    # Wait for strike selector to have options
    try:
        page.wait_for_selector("#strikeSelector option", timeout=5000)
    except:
        print("Strike selector still empty")

    strikes = page.locator("#strikeSelector option")
    print(f"Strike options: {strikes.count()}")

    # Check timezone on chart (hard to check text in canvas, but we can check if charts initialized)
    is_chart_init = page.evaluate("() => !!window.prodesk.charts.price.index")
    print(f"Index Chart Initialized: {is_chart_init}")

    page.screenshot(path="/home/jules/verification/modern_final_debug.png")

if __name__ == "__main__":
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        try:
            test_modern_features(page)
        except Exception as e:
            print(f"Error: {e}")
        finally:
            browser.close()

File: verify_final.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();

  page.on('console', msg => console.log('BROWSER CONSOLE:', msg.text()));
  page.on('pageerror', err => {
    console.log('BROWSER PAGE ERROR:', err.message);
    process.exit(1);
  });

  try {
    console.log('Navigating to /modern...');
    await page.goto('http://localhost:3000/modern', { waitUntil: 'networkidle', timeout: 60000 });

    console.log('Checking for syntax errors (presence of main elements)...');
    const title = await page.title();
    console.log('Page title:', title);

    const header = await page.waitForSelector('header', { timeout: 10000 });
    if (header) {
        console.log('Dashboard loaded successfully, no blocking JS errors found.');
    }

    await page.screenshot({ path: 'final_verification.png', fullPage: true });
    console.log('Screenshot saved to final_verification.png');

  } catch (error) {
    console.error('Verification failed:', error);
    process.exit(1);
  } finally {
    await browser.close();
  }
})();
